tests/check/check_userset.go

```diff
@@ -997,4 +997,73 @@ var usersetCompleteTestingModelTest = []*stage{
 			},
 		},
 	},
+	{
+		Name: "usersets_tuple_cycle2",
+		Tuples: []*openfgav1.TupleKey{
+			// user exists so no cycle
+			{Object: "directs-user:utc2_1", Relation: "tuple_cycle2", User: "user:utc2_1"},
+			{Object: "ttus:utc2_1", Relation: "direct_parent", User: "directs-user:utc2_1"},
+			{Object: "usersets-user:utc2_1", Relation: "tuple_cycle2", User: "ttus:utc2_1#tuple_cycle2"},
+			{Object: "directs-user:utc2_2", Relation: "tuple_cycle2", User: "usersets-user:utc2_1#tuple_cycle2"},
```

Comment:
I think to actually make it a full cycle (but with exit), you need 
```
{Object: "directs-user:utc2_1", Relation: "tuple_cycle2", User: "usersets-user:utc2_1#tuple_cycle2"},
```

See https://github.com/openfga/openfga/blob/c2ef51f0c3fd80a62bd7ceed29360de5984b73d5/tests/check/complexity_three.go#L758-L763 as example